x-pack/plugins/endpoint/public/embeddables/resolver/store/actions.ts

```diff
@@ -43,14 +43,15 @@ interface UserChangedSelectedEvent {
 interface AppRequestedResolverData {
   readonly type: 'appRequestedResolverData';
 }
+
 /**
  * When the user switches the active descendent of the Resolver.
  */
 interface UserFocusedOnResolverNode {
   readonly type: 'userFocusedOnResolverNode';
   readonly payload: {
     /**
-     * Used to identify the process node that should be brought into view.
+     * Used to identify the process node that the user focused on (in the DOM)
```

Comment:
@oatkiller this might actually not be true... e.g. when someone switches the active descendant (with the Panel), the tree node doesn't have/get focus until it's returned to its parent composite

Reply:
I don't understand what you mean. But if you have a better suggestion for this, please let me know. Resolver already has a concept of 'focusing' on a process. It happens when you click the icon in the panel and the graph moves to center the process. This action should probably be combined with that one. Thoughts?

Reply:
According to the `role` description for tree, `focus` and `activedescendant` are independent concepts. When the user clicks the icon in the panel, the _icon_ should still have input focus, but the Resolver's `aria-activedescendant` changes to the node's ID and when focus returns to the tree/(Resolver) you assign focus back to the `activedescendant` node.

With this is mind, my suggestion might be to change this interface name to something like `userSelectedResolverNode` but we're going to be taking a swipe at this whole active/focus thing soon so feel free to let it ride for now.

Reply:
Maybe I understand. Basically you're saying that the 'selected' (our definition) resolver node and the focused (DOM's definition) element aren't necessarily (and shouldn't be) the same?